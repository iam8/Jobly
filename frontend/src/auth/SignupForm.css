.SignupForm {
    height: calc(100vh - 50px);
    display: flex;
    flex-direction: column;
    align-items: center;
}

.SignupForm-heading {
    padding-top: 3rem;
    padding-bottom: 1rem;
    color: white;
    text-shadow: 2px 2px 3px black;
}

.SignupForm-card {
    width: 30%;
}

.SignupForm label {
    font-weight: bold;
}