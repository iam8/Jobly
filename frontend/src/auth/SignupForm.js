import React, {useState} from "react";
import { useHistory } from "react-router-dom";
import { Alert, Button, Card, CardBody, Form, FormGroup, Label, Input } from "reactstrap";

import "./SignupForm.css";


/**
 * Route: /signup
 *
 * Form used to register a new user on the Jobly website.
 *
 * Manages state updates on changes to the form inputs.
 *
 * On successful sign up, calls signup (function prop) and redirects to / (home).
 *
 * On signup failure, displays alert and error message.
 */
function SignupForm({signup}) {
    const INIT_FORM = {
        username: "",
        password: "",
        firstName: "",
        lastName: "",
        email: ""
    };

    const [formData, setFormData] = useState(INIT_FORM);
    const [formErrors, setFormErrors] = useState([]);
    const history = useHistory();

    /** Handle changes to inputs. */
    const handleChange = (evt) => {
        const {name, value} = evt.target;
        setFormErrors([]);
        setFormData((formData) => ({
            ...formData,
            [name]: value
        }));
    }

    /** Form submission - register user and redirect to homepage on success. */
    const handleSubmit = async (evt) => {
        evt.preventDefault();
        const result = await signup(formData);

        if (result.success) {
            history.push("/"); // Redirect to homepage
        } else {
            setFormErrors(result.err);
        }
    }

    /** Show an alert message on failure to sign up. */
    const renderAlert = () => {
        if (formErrors.length) {
            return (
                <Alert color="danger">
                    Sign up failed - {formErrors}
                </Alert>
            );
        }
    }

    return (
        <div className="SignupForm">
            <h2 className="SignupForm-heading">Sign Up</h2>

            <Card className="SignupForm-card">
                <CardBody>
                    <Form className="SignupForm-form" onSubmit={handleSubmit}>
                        <FormGroup>
                            <Label htmlFor="username">Username</Label>
                            <Input
                                id="username"
                                name="username"
                                type="text"
                                value={formData.username}
                                onChange={handleChange}
                            />
                        </FormGroup>

                        <FormGroup>
                            <Label htmlFor="password">Password</Label>
                            <Input
                                id="password"
                                name="password"
                                type="password"
                                value={formData.password}
                                onChange={handleChange}
                            />
                        </FormGroup>

                        <FormGroup>
                            <Label htmlFor="first-name">First name</Label>
                            <Input
                                id="first-name"
                                name="firstName"
                                type="text"
                                value={formData.firstName}
                                onChange={handleChange}
                            />
                        </FormGroup>

                        <FormGroup>
                            <Label htmlFor="last-name">Last name</Label>
                            <Input
                                id="last-name"
                                name="lastName"
                                type="text"
                                value={formData.lastName}
                                onChange={handleChange}
                            />
                        </FormGroup>

                        <FormGroup>
                            <Label htmlFor="email">Email</Label>
                            <Input
                                id="email"
                                name="email"
                                type="email"
                                value={formData.email}
                                onChange={handleChange}
                            />
                        </FormGroup>

                        {renderAlert()}

                        <Button color="primary" block>Submit</Button>
                    </Form>
                </CardBody>
            </Card>
        </div>
    )
}


export default SignupForm;
