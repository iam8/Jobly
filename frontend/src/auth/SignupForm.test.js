import React from "react";
import { render } from "@testing-library/react";
import SignupForm from "./SignupForm";


test("Renders without crashing", () => {
    render(<SignupForm />);
})


test("Matches snapshot", () => {
    const { asFragment } = render(
        <SignupForm />
    );

    expect(asFragment()).toMatchSnapshot();
})