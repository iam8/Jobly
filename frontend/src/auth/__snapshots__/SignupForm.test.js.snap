// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <h2
      class="SignupForm-heading"
    >
      Sign Up
    </h2>
    <div
      class="SignupForm-card card"
    >
      <div
        class="card-body"
      >
        <form
          class="SignupForm-form"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="username"
            >
              Username
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="first-name"
            >
              First name
            </label>
            <input
              class="form-control"
              id="first-name"
              name="firstName"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="last-name"
            >
              Last name
            </label>
            <input
              class="form-control"
              id="last-name"
              name="lastName"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
              Email
            </label>
            <input
              class="form-control"
              id="email"
              name="email"
              type="email"
              value=""
            />
          </div>
          <button
            class="btn btn-primary d-block w-100"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
